04-sheets-script/sheets_script.js

```diff
@@ -0,0 +1,102 @@
+function fetchDataAndPopulateSheet() {
+  const sheet = SpreadsheetApp.getActiveSpreadsheet().getActiveSheet();
+  const loadingCell = sheet.getRange("D1");
+  loadingCell.setValue("Bear with us... asking Contextual AI...");
+
+//  update with your agent ID and API key
+  const apiUrl = 'https://api.contextual.ai/v1/agents/99bd0fc6-75ca-4057-91b1-28dc1c201221/query?retrievals_only=false'
```

Comment:
Should this be a hard coded Agent ID?

Reply:
it's not a real Agent ID, I scrambled the numbers. I wanted to make it easy so someone could cut and paste their agent Id into the URL.  Open to finding a nicer way to do this.

Reply:
Is there such a thing as "SECRETS" like in google Colab? Or embed the agent id in the sheet somehow?

Reply:
this is all markdown and javascript, so not sure how colab would fit in -- we could just remove it and replace it with "ADD_AGENT_ID_HERE" - but since its in a long URL, its not obvious that most people would get it right